### PR TITLE
chore: new progress bar implement

### PR DESCRIPTION
**Pull request by Siglud**

### CODE REVIEW

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/component/driver/deploy/azure/azureStorageDeployDriver.ts [hunk 1]
```diff
@@ -191,10 +176,7 @@ export class AzureStorageDeployDriverImpl extends AzureDeployImpl {
     return (blob.properties.contentLength ?? -1) > 0;
   }
 
-  createProgressBar(ui?: UserInteraction): IProgressHandler | undefined {
-    return ui?.createProgressBar(
-      `Deploying ${this.workingDirectory ?? ""} to Azure Storage Service`,
-      this.progressNames.length
-    );
+  createProgressBar() {
```
Comment (tecton) [inline packages/fx-core/src/component/driver/deploy/azure/azureStorageDeployDriver.ts:179]: Is this function invoked only 1 time?

File: packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts [hunk 2]
```diff
@@ -60,7 +59,7 @@ export abstract class BaseDeployImpl {
 
   async run(): Promise<ExecutionResult> {
     await this.context.logProvider.debug("start deploy process");
-
+    this.createProgressBar();
```
Comment (tecton) [inline packages/fx-core/src/component/driver/deploy/azure/impl/baseDeployImpl.ts:62]: You can also use `updateProgressbar` middleware

File: packages/fx-core/src/component/driver/script/scriptDriver.ts [hunk 3]
```diff
@@ -25,6 +25,9 @@ export class ScriptDriver implements StepDriver {
   @hooks([addStartAndEndTelemetry(ACTION_NAME, TelemetryConstant.SCRIPT_COMPONENT)])
   async run(args: unknown, context: DriverContext): Promise<Result<Map<string, string>, FxError>> {
     const typedArgs = args as ScriptDriverArgs;
+    await context.progressBar?.next(
+      `Run command ${typedArgs.run} at ${typedArgs.workingDirectory}`
```
Comment (tecton) [inline packages/fx-core/src/component/driver/script/scriptDriver.ts:29]: Can you localize this string?

**Review by tecton — COMMENTED** [review]
LGTM, please also remove unused strings in `package.nls.json`.

File: packages/fx-core/src/component/driver/deploy/azure/azureStorageDeployDriver.ts (hunk 1 above)
Comment (Siglud) [reply packages/fx-core/src/component/driver/deploy/azure/azureStorageDeployDriver.ts:179]: Yes, I will rename it to 'updateProgressbar' for clarity.

File: packages/fx-core/src/component/driver/script/scriptDriver.ts (hunk 3 above)
Comment (Siglud) [reply packages/fx-core/src/component/driver/script/scriptDriver.ts:29]: > LGTM, please also remove unused strings in `package.nls.json`.

Some of the strings are still needed by the v2 version.

**Review by hund030 — APPROVED** [review]

**Review by jidddddd — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by eriolchan — APPROVED** [review]

**Pull request merged**